hazelcast/src/main/java/com/hazelcast/cache/impl/nearcache/impl/store/NearCacheDataRecordStore.java

```diff
@@ -157,4 +167,18 @@ public int size() {
         return store.size();
     }
 
+    @Override
+    public void doExpiration() {
+        for (Map.Entry<K, NearCacheDataRecord> entry : store.entrySet()) {
+            if (Thread.currentThread().isInterrupted()) {
```

Comment:
Yes, while destroying near-cache, expiration task is shutdown via `shutdown(true)` and this also interrupts `ExpirationTask` runner thread. So via `Thread.currentThread().isInterrupted()`, we don't wait until expiration finished since near-cache is destroyed and no need to wait for finishing of expiration.